File: backend/app/models/project.py
from pydantic import BaseModel
from typing import Optional
from enum import Enum


class CategoryEnum(str, Enum):
    GAUNTLET = "gauntlet"
    OTHER = "other"


class Project(BaseModel):
    id: str
    name: str
    description: str
    long_description: Optional[str] = None
    category: CategoryEnum
    week: Optional[int] = None
    tech_stack: list[str]
    repo_url: Optional[str] = None
    live_url: Optional[str] = None
    writeup_url: Optional[str] = None
    thumbnail: Optional[str] = None
    screenshots: list[str] = []
    icon: str = "folder"
    created_at: str
    featured: bool = False
    challenges: Optional[str] = None
    learnings: Optional[str] = None


class ProjectListResponse(BaseModel):
    projects: list[Project]
    total: int


class CategoryResponse(BaseModel):
    id: str
    name: str
    description: str
    icon: str

File: frontend/src/types/project.ts
export type ProjectCategory = 'gauntlet' | 'other';

export interface SubFolder {
  name: string;
  subFolders?: SubFolder[];
  liveUrl?: string;
  repoUrl?: string;
}

export interface Project {
  id: string;
  name: string;
  description: string;
  longDescription?: string;
  category: ProjectCategory;
  week?: number;
  techStack: string[];
  repoUrl?: string;
  liveUrl?: string;
  demoUrl?: string;
  vizUrl?: string;
  writeupUrl?: string;
  company?: string;
  video?: string;
  thumbnail?: string;
  screenshots?: string[];
  icon: string;
  createdAt: string;
  featured: boolean;
  highlights?: string[];
  challenges?: string;
  learnings?: string;
  spec?: string;
  passwordProtected?: boolean;
  subFolders?: SubFolder[];
}

export interface Category {
  id: string;
  name: string;
  description: string;
  icon: string;
}
